core/trino-main/src/main/java/io/trino/server/protocol/spooling/SegmentResource.java

```diff
@@ -72,26 +72,29 @@ public Response download(@Context UriInfo uriInfo, @PathParam("identifier") Stri
             throws IOException
     {
         SpooledSegmentHandle handle = handle(identifier, headers);
-        Optional<DirectLocation> directLocation = spoolingManager.directLocation(handle);
 
-        // Direct access is enabled & supported - redirect user to the spooled location using pre-signed URIs
-        if (directLocation.isPresent()) {
-            return Response
-                    .seeOther(directLocation.get().uri())
+        return switch (retrievalMode) {
+            case DIRECT -> throw new ServiceUnavailableException("Retrieval mode is DIRECT but segment resource was called");
+            case COORDINATOR_PASS_THROUGH -> Response.ok(spoolingManager.openInputStream(handle)).build();
+            case COORDINATOR_WORKERS_PASS_THROUGH -> {
```

Comment:
nit; passthrough is a single word. Drop underscore

Reply:
no, it's written as `pass-through` (I've checked that in the dictonary)